### PR TITLE
migrate Data subpackage

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR deletes all the files associated to data as the package was moved to a standalone repo:  https://github.com/Lightning-AI/litdata

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @tchaton

### CODE REVIEW

**Review by yurijmikhalevich — APPROVED** [review]

File: requirements/data/data.txt
Comment (carmocca) [inline requirements/data/data.txt:1]: I think you want to remove every other requirement?

File: .github/workflows/ci-tests-data.yml
Comment (carmocca) [inline .github/workflows/ci-tests-data.yml:1]: Can you also remove https://github.com/Lightning-AI/pytorch-lightning/blob/f2f3ef5d3dfde71e8e1618f3ba49ff8cd82ef7e8/.github/checkgroup.yml#L153-L175?

**Review by carmocca — COMMENTED** [review]
I suggest grepping the entire project for any leftover usages (I haven't checked if there are any)

File: requirements/data/data.txt
Comment (tchaton) [reply requirements/data/data.txt:1]: done.

File: .github/workflows/ci-tests-data.yml
Comment (tchaton) [reply .github/workflows/ci-tests-data.yml:1]: done.

File: src/lightning/__setup__.py [hunk 1]
```diff
@@ -48,8 +48,7 @@ def _prepare_extras() -> Dict[str, Any]:
     extras["app-extra"] = extras["app-app"] + extras["app-cloud"] + extras["app-ui"] + extras["app-components"]
     extras["app-all"] = extras["app-extra"]
     extras["app-dev"] = extras["app-all"] + extras["app-test"]
-    extras["data-all"] = extras["data-data"] + extras["data-cloud"] + extras["data-examples"]
-    extras["data-dev"] = extras["data-all"] + extras["data-test"]
+    extras["data-all"] = extras["data-data"]
```
Comment (carmocca) [inline src/lightning/__setup__.py:51]: I don't see a point in keeping this?

```suggestion
```

**Review by carmocca — APPROVED** [review]

File: src/lightning/__setup__.py (hunk 1 above)
Comment (tchaton) [reply src/lightning/__setup__.py:51]: @carmocca We still want `pip install -e '.[data]'` to work.

Comment (tchaton) [reply src/lightning/__setup__.py:51]: As `lightning.data` is still available.

Comment (tchaton) [reply src/lightning/__setup__.py]: Ok, removed it.

**Review by awaelchli — APPROVED** [review]

File: requirements/data/data.txt [hunk 2]
```diff
@@ -1,8 +1,3 @@
-# NOTE: the upper bound for the package version is only set for CI stability, and it is dropped while installing this package
-#  in case you want to preserve/enforce restrictions on the latest compatible version, add "strict" as an in-line comment
+# NOTE: this is here only to expose `pip install lightning[data]`. we don't install or test it in this project's CI
 
-lightning-utilities >=0.8.0, <0.10.0
-# to be able to include also PL 2.0 and preserve `>` needed for CI min version bypass
-torch >0.14.0, <=2.2.0
-lightning-cloud
-filelock
+litdata >= 0.2.0rc
```
Comment (Borda) [inline requirements/data/data.txt:3]:
```suggestion
litdata >= 0.2.0rc, <0.3.0
```

File: .github/workflows/ci-pkg-install.yml [hunk 3]
```diff
@@ -85,6 +85,7 @@ jobs:
         working-directory: .actions/
         run: |
           pip install -q pytest -r requirements.txt
+          pip install -U litdata
```
Comment (Borda) [inline .github/workflows/ci-pkg-install.yml:88]:
```suggestion
```

**Review by Borda — CHANGES_REQUESTED** [review]

**Review by Borda — APPROVED** [review]

**Review by lantiga — APPROVED** [review]
LGTM

**Pull request merged**